sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCSVScan.scala

```diff
@@ -168,7 +168,8 @@ object GpuCSVScan {
         GpuCsvUtils.dateFormatInRead(parsedOptions), parseString = true)
     }
 
-    if (types.contains(TimestampType)) {
+    // Date type needs to be checked even it has no time zone info as its fields.
+    if (types.contains(TimestampType) || types.contains(DateType)) {
```

Comment:
Is this because the timezone might be in the date text? I'd like to understand this better.

Reply:
For date type, timezone needs to be checked also. This is because JVM timezone is used to get days offset before rebasing Julian to Gregorian in Spark while not in Rapids. 

In details, for CSV data format, Spark uses dateFormatter to parse string as date data type which utilizes [[org.apache.spark.sql.catalyst.DateFormatter]]. And CSV format (e.g., [[UnivocityParser]]), it uses [[LegacyFastDateFormatter]] which is based on Apache Commons FastDateFormat. It parse string into Java util.Date base on JVM default timezone. From Java util.Date, it's converted into java.sql.Date type. By leveraging [[JavaDateTimeUtils]], it finally do `rebaseJulianToGregorianDays` considering its offset to UTC timezone.

Code comments added inline as well.